bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/internal/discovery/OpenWebNetDeviceDiscoveryService.java

```diff
@@ -0,0 +1,164 @@
+        bridgeUID = handler.getThing().getUID();
+        logger.debug("Constructor for bridge: '{}'", bridgeUID);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return OpenWebNetDeviceDiscoveryService.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    protected void startScan() {
+        logger.info("------ startScan() - SEARCHING for DEVICES on bridge '{}' ({}) ...",
+                bridgeHandler.getThing().getLabel(), bridgeUID);
+        bridgeHandler.searchDevices();
+    }
+
+    @Override
+    protected void stopScan() {
+        logger.debug("------ stopScan() on bridge '{}'", bridgeUID);
+        bridgeHandler.scanStopped();
+    }
+
+    @Override
+    public void abortScan() {
+        logger.debug("------ abortScan() on bridge '{}'", bridgeUID);
+        bridgeHandler.scanStopped();
+    }
+
+    /**
+     * Create and notify to Inbox a new DiscoveryResult based on WHERE, OpenDeviceType and BaseOpenMessage
+     *
+     * @param where the discovered device's address (WHERE)
+     * @param deviceType {@link OpenDeviceType} of the discovered device
+     * @param message the OWN message received that identified the device (optional)
+     */
+    public void newDiscoveryResult(Where where, OpenDeviceType deviceType, @Nullable BaseOpenMessage baseMsg) {
+        logger.info("newDiscoveryResult() WHERE={}, deviceType={}", where, deviceType);
+        ThingTypeUID thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_GENERIC_DEVICE; // generic device
+        String thingLabel = OpenWebNetBindingConstants.THING_LABEL_GENERIC_DEVICE;
+        Who deviceWho = Who.UNKNOWN;
+        switch (deviceType) {
+            case ZIGBEE_ON_OFF_SWITCH: {
+                thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_ZB_ON_OFF_SWITCH;
+                thingLabel = OpenWebNetBindingConstants.THING_LABEL_ZB_ON_OFF_SWITCH;
+                deviceWho = Who.LIGHTING;
+                break;
+            }
+            case ZIGBEE_DIMMER_SWITCH: {
+                thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_ZB_DIMMER;
+                thingLabel = OpenWebNetBindingConstants.THING_LABEL_ZB_DIMMER;
+                deviceWho = Who.LIGHTING;
+                break;
+            }
+            case SCS_ON_OFF_SWITCH: {
+                thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_BUS_ON_OFF_SWITCH;
+                thingLabel = OpenWebNetBindingConstants.THING_LABEL_BUS_ON_OFF_SWITCH;
+                deviceWho = Who.LIGHTING;
+                break;
+            }
+            case SCS_DIMMER_SWITCH: {
+                thingTypeUID = OpenWebNetBindingConstants.THING_TYPE_BUS_DIMMER;
+                thingLabel = OpenWebNetBindingConstants.THING_LABEL_BUS_DIMMER;
+                deviceWho = Who.LIGHTING;
+                break;
+            }
+            default: {
+                logger.warn("Device type {} is not supported, default to GENERIC device (WHERE={})", deviceType, where);
```

Comment:
If I understand correctly, your're adding generic devices you can't control or receive information for. What's the purpose of these?

Reply:
Sometime devices are there but not recognised correctly, other times since they are not powered they are found by the gateway but their type is not recognised. By creating such GENERIC Things helps the user and the programmer to understand better the situation. I think something similar happens for the ZigBee binding, where unrecognised devices are still found but not controllable.